### PR TITLE
Fix/refactor app

### PR DESCRIPTION
**Pull request by itsnameissteven**
## Is this a fix, feature, or refactor?
This is a refactor

## What is the change?
* A new main component has been built to handle main page view
* Moved error handling to the utilities.js file
* Fixed linting errors
* Adjusted state when clicking home button

## What does it fix?
This refactors the render method in the app component.  
Conditional rendering now happens in Main.js 
A method has been built to display errors or main page based on fetch errors being present.

## Where should the reviewer start?
Review the new Main.Js file then utilities.js to see new error handling function.

## How should this be tested?
Test in the browser. Uncomment out the last .then method in displayMovieDetails and click on movie posters to see the error message on the web  app.

### CODE REVIEW

**Comment by itsnameissteven** [comment]
@sertmer
@letakeane

File: src/App.js [hunk 1]
```diff
@@ -22,87 +20,74 @@ class App extends Component {
...
-          throw new Error();
-        }
-
-        return response.json();
+        return handleErrors(response);
       })
       .then((movieData) => this.setState({ selectedMovieDetails: movieData.movie }))
-      .catch((err) => this.setState( {fetchError: true }));
+      .catch(() => this.setState({ fetchError: true }))
+      // .then(this.setState({ fetchError: true, fetchStatus: 404 })) //to test errors on click
```
Comment (knees4bees) [inline src/App.js:38]: I think we won't need this line anymore once we get Cypress testing working

File: src/App.js [hunk 2]
```diff
@@ -22,87 +20,74 @@ class App extends Component {
...
       default:
         message = 'Oops! Something went wrong. Please try again.';
     }
 
+    return (
+      <Error message={message} />
+    );
+  }
+
+  toggleView = () => {
```
Comment (knees4bees) [inline src/App.js:71]: I like this refactor a lot. It makes sense to have a `<Main />` component with conditional rendering.

**Review by knees4bees — APPROVED** [review]

**Review by letakeane — COMMENTED** [review]
Nice work creating that Main component and making your apiCalls/utils file more robust! Looking nice so far.

**Pull request merged**